### PR TITLE
Add preflight package (#149)

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Preflight package implements should implement
all preflight functionality per one Module.
Current commit will only implement image verification: image presence and correct kernel version

[MGMT-11438](https://issues.redhat.com//browse/MGMT-11438)

### CODE REVIEW

**Comment by yevgeny-shnaidman** [comment]
Upstream PR for cherry-pick:
https://github.com/kubernetes-sigs/kernel-module-management/commit/0e84a4e812a4e2fca0e80a18168207c1660277c5

File: internal/module/kernelmapper.go [hunk 1]
```diff
@@ -19,14 +19,11 @@ const (
 )
 
 type NodeOSConfig struct {
-	kernelFullVersion  string `subst:"KERNEL_FULL_VERSION"`
-	kernelVersionMMP   string `subst:"KERNEL_XYZ"`
-	kernelVersionMajor string `subst:"KERNEL_X"`
-	kernelVersionMinor string `subst:"KERNEL_Y"`
-	kernelVersionPatch string `subst:"KERNEL_Z"`
-	kernelRHELVersion  string `subst:"KERNEL_RHEL_VERSION"`
```
Comment (ybettan) [inline internal/module/kernelmapper.go:27]: Don't we need RHEL's  info here?

Comment (yevgeny-shnaidman) [reply internal/module/kernelmapper.go:27]: i am currently just backporting. Once i have the basics in, i will need to add changes to the downstream, and that will include all the needed changes

Comment (ybettan) [reply internal/module/kernelmapper.go:27]: Ack.

**Pull request merged**